domain/src/main/kotlin/no/nav/su/se/bakover/domain/behandling/Behandlingsinformasjon.kt

```diff
@@ -0,0 +1,244 @@
+            uførhet,
+            flyktning,
+            lovligOpphold,
+            fastOppholdINorge,
+            oppholdIUtlandet,
+            formue,
+            personligOppmøte,
+            sats
+        ).all { it != null && it.isValid() && it.isComplete() }
+
+    fun isInnvilget() =
+        isComplete() &&
+            listOf(
+                uførhet?.status == Uførhet.Status.VilkårOppfylt,
+                flyktning?.status == Flyktning.Status.VilkårOppfylt,
+                lovligOpphold?.status == LovligOpphold.Status.VilkårOppfylt,
+                fastOppholdINorge?.status == FastOppholdINorge.Status.VilkårOppfylt,
+                oppholdIUtlandet?.status == OppholdIUtlandet.Status.SkalHoldeSegINorge,
+                formue?.status == Formue.Status.Ok,
+                personligOppmøte?.status.let {
+                    it == PersonligOppmøte.Status.FullmektigMedLegeattest ||
+                        it == PersonligOppmøte.Status.MøttPersonlig ||
+                        it == PersonligOppmøte.Status.Verge
+                }
+            ).all { it }
+
+    fun isAvslag() =
+        listOf(
+            uførhet?.status == Uførhet.Status.VilkårIkkeOppfylt,
+            flyktning?.status == Flyktning.Status.VilkårIkkeOppfylt,
+            lovligOpphold?.status == LovligOpphold.Status.VilkårIkkeOppfylt,
+            fastOppholdINorge?.status == FastOppholdINorge.Status.VilkårIkkeOppfylt,
+            oppholdIUtlandet?.status == OppholdIUtlandet.Status.SkalVæreMerEnn90DagerIUtlandet,
+            personligOppmøte?.status.let {
+                it == PersonligOppmøte.Status.FullmektigUtenLegeattest ||
+                    it == PersonligOppmøte.Status.IkkeMøttOpp
+            }
+        ).any { it }
+
+    abstract class Base {
+        abstract fun isValid(): Boolean
+        abstract fun isComplete(): Boolean
+    }
+
+    data class Uførhet(
+        val status: Status,
+        val uføregrad: Int?,
+        val forventetInntekt: Int?
+    ) : Base() {
+        enum class Status {
+            VilkårOppfylt,
+            VilkårIkkeOppfylt,
+            HarUføresakTilBehandling
+        }
+
+        override fun isValid(): Boolean =
+            when (status) {
+                Status.VilkårOppfylt -> uføregrad != null && forventetInntekt != null
+                Status.VilkårIkkeOppfylt -> uføregrad == null && forventetInntekt == null
+                Status.HarUføresakTilBehandling -> uføregrad != null && uføregrad > 0 && forventetInntekt != null && forventetInntekt > 0
+            }
+
+        override fun isComplete(): Boolean = isValid()
+    }
+
+    data class Flyktning(
+        val status: Status,
+        val begrunnelse: String?
+    ) : Base() {
+        enum class Status {
+            VilkårOppfylt,
+            VilkårIkkeOppfylt,
+            Uavklart
+        }
+
+        override fun isValid(): Boolean =
+            when (status) {
+                Status.VilkårOppfylt -> begrunnelse == null
```

Comment:
Skal det være en != her? Hm, eller kanskje ikke? Kommer kanskje an på om det er lov å si "Oppfylt" uten å ha en begrunnelse. Hvis den må være angitt så gjelder det på litt flere nedover.

```suggestion
                Status.VilkårOppfylt -> begrunnelse != null
```

Reply:
Disse reglene er egentlig bare basert på halvkvalifisert gjetning. Litt usikker på om Agnethe/fag har kommet så langt som at de har helt klarhet i hva som skal være påkrevd når.

Reply:
Ja, sant. Tror det er like så greit å bare få det ut og så tilpasse når vi ser det i bruk. Casen med påkrevd begrunnelse ved ja er jo også litt odde, tror vi bare trenger den så lenge vi ikke viser grunnlagsdata i løsningen (altså info fra integrasjoner mot andre, f.eks. status på uføre).